tools/jobs.php

```diff
@@ -0,0 +1,420 @@
+    {
+        return $this->parameterList;
+    }
+
+    protected function getParameterValue(string $parameter, string $default = null): ?string
+    {
+        if (!isset($this->getParameterList()[$parameter])) {
+            return $default;
+        }
+
+        return $this->getParameterList()[$parameter];
+    }
+
+    /**
+     * Print command usage information.
+     */
+    public function usage()
+    {
+        $this->getCommandInterface()->line('<comment>Usage:</comment>');
+        $this->getCommandInterface()->line('command [arguments]' . PHP_EOL);
+        $this->getCommandInterface()->line('<comment>Available commands for the "jobs" namespace:</comment>');
+
+        $width = $this->getColumnWidth(array_keys(self::AVAILABLE_OPTIONS));
+
+        foreach (self::AVAILABLE_OPTIONS as $commandName => $description) {
+            $spacingWidth = $width - Helper::width($commandName);
+            $this->getCommandInterface()->line(
+                sprintf(
+                    '  <info>%s</info>%s%s',
+                    $commandName,
+                    str_repeat(' ', $spacingWidth),
+                    $description
+                )
+            );
+        }
+    }
+
+    /**
+     * Alias for usage command
+     */
+    public function help(): void
+    {
+        $this->usage();
+    }
+
+    /**
+     * Retrieve the columnWidth based on the commands text size
+     */
+    protected function getColumnWidth(array $commands): int
+    {
+        $widths = [];
+
+        foreach ($commands as $command) {
+            $widths[] = Helper::width($command);
+        }
+
+        return $widths ? max($widths) + 2 : 0;
+    }
+
+    /**
+     * List all queued jobs
+     */
+    protected function list(): void
+    {
+        $perPage = $this->getParameterValue('perPage', '10');
+        $page = $this->getParameterValue('page', '1');
+
+        $this->total();
+
+        $data = Repo::job()
+            ->setOutputFormat(Repo::job()::OUTPUT_CLI)
+            ->perPage((int) $perPage)
+            ->setPage((int) $page)
+            ->showQueuedJobs();
+
+        $this->getCommandInterface()
+            ->table(
+                [
+                    [
+                        new TableCell(
+                            'Queued Jobs',
+                            [
+                                'colspan' => 7,
+                                'style' => new TableCellStyle(['align' => 'center'])
+                            ]
+                        )
+                    ],
+                    [
+                        'ID',
+                        'Queue',
+                        'Job Display Name',
+                        'Attempts',
+                        'Reserved At',
+                        'Available At',
+                        'Created At'
+                    ]
+                ],
+                $data->all(),
+            );
+
+        $pagination = [
+            'pagination' => [
```

Comment:
The `0`, `1`, and `2` magic numbers correspond to last page, current page, and next page (where available), right? Could you use class constants for these?